scaleway/resource_k8s_cluster.go

```diff
@@ -157,6 +158,13 @@ func resourceScalewayK8SCluster() *schema.Resource {
 				Required:    true,
 				Description: "Delete additional resources like block volumes and loadbalancers on cluster deletion",
 			},
+			"private_network_id": {
```

Comment:
As this is an ID that can be passed from another regional resource, you can add `customizeDiffLocalityCheck` to your CustomizeDiff.

```
		CustomizeDiff: customdiff.All(
			func(ctx context.Context, diff *schema.ResourceDiff, i interface{}) error {
				autoUpgradeEnable, okAutoUpgradeEnable := diff.GetOkExists("auto_upgrade.0.enable")

				version := diff.Get("version").(string)
				versionIsOnlyMinor := len(strings.Split(version, ".")) == 2

				if okAutoUpgradeEnable && versionIsOnlyMinor != autoUpgradeEnable.(bool) {
					return fmt.Errorf("minor version x.y must be used with auto upgrade enabled")
				}

				return nil
			},
			customizeDiffLocalityCheck("private_network_id"),
		),
```

It will make the plan fail if private network is in another region:
```hcl
resource scaleway_vpc_private_network "pn" {
  zone = "nl-ams-1"
}

resource scaleway_k8s_cluster "cluster" {
  region = "fr-par"
  private_network_id = scaleway_vpc_private_network.pn.id
}

>>> given private_network_id <zone/uuid> has different locality than the resource "fr-par"
```